swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -170,17 +170,36 @@ public List<Contract> getContracts(
   }
 
   /**
-   * Delete a contract for a given uuid. This is hard delete, because its intended use is for
-   * cleaning up test data.
+   * Delete a contract for a given uuid. This is soft delete. It sets the end date of a contract to
+   * the current timestamp.
    *
    * @param uuid
    */
   @Transactional
   public void deleteContract(String uuid) {
 
-    var isSuccessful = contractRepository.deleteById(UUID.fromString(uuid));
+    updateContractEndDate(uuid, OffsetDateTime.now());
+  }
+
+  /**
+   * Look up a contract via uuid and update the record to have the given end date.
+   *
+   * @param uuid
+   * @param endDate
+   */
+  @Transactional
+  Optional<ContractEntity> updateContractEndDate(String uuid, OffsetDateTime endDate) {
+    var contract = contractRepository.findContract(UUID.fromString(uuid));
+
+    if (Objects.nonNull(contract)) {
+      log.debug("Setting end date to {} for contract {}", endDate, uuid);
+      contract.setEndDate(endDate);
+      contract.setLastUpdated(OffsetDateTime.now());
```

Comment:
I was hoping to set the same endDate here since the enddate and lastDate gets updated at same time.